src/fen.rs

```diff
@@ -112,10 +113,22 @@ fn en_passant_pos(input: &str) -> IResult<&str, Option<Pos>> {
     alt((value(None, tag("-")), map(pos, Some)))(input)
 }
 
-// TODO
-fn castling(input: &str) -> IResult<&str, ()> {
-    let (input, _res) = many1(alt((tag("-"), alpha1)))(input)?;
-    Ok((input, ()))
+fn castling(input: &str) -> IResult<&str, Castling> {
+    let (input, res) = many1(alt((tag("-"), tag("K"), tag("Q"), tag("k"), tag("q"))))(input)?;
```

Comment:
is `input` being shadowed here?

Reply:
yep. i think this is a case where it's better to shadow input, because you probably don't want the old reference to it. it's a technique used in the nom documentation too: https://docs.rs/nom/5.1.1/nom/